src/components/SearchResultsCount.js

```diff
@@ -2,50 +2,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+type SearchResultsCountProps = {
+  /**
+   * A response can be passed in from custom searches if you don't want
+   *  to use the response on the searcher in this.context
+   */
+  response: QueryResponse | null;
+  /**
+   * If attempting to pass in a custom response, there may be queries that result in errors,
+   * which can be passed in here. Otherwise the error from the searcher
+   * on this.context will be used.
+   */
+  error: string | null;
+}
+
+type SearchResultsCountDefaultProps = {
+  response: QueryResponse | null;
+  error: string | null;
+}
+
 /**
  * The count of search results or an indication
  * that a search hasn't yet happened.
  */
-export default class SearchResultsCount extends React.Component<void, {}, void> {
+export default class SearchResultsCount extends React.Component<SearchResultsCountProps, SearchResultsCountDefaultProps, void> {
+  static defaultProps = {
+    response: null,
+    error: null,
+  }
+
   static contextTypes = {
     searcher: PropTypes.any,
   };
 
   static displayName = 'SearchResultsCount';
 
   render() {
-    const searcher = this.context.searcher;
+    let response;
     let message;
     let countMessage;
-    if (searcher) {
-      const response = searcher.state.response;
-      if (response) {
-        const count = response.totalHits;
-        if (count === 0) {
-          countMessage = 'No results found';
-        } else if (count === 1) {
-          countMessage = '1 result found';
-        } else {
-          const countStr = Number(count).toLocaleString();
-          countMessage = `${countStr} results found`;
-        }
-        message = (
-          <span>
-            {countMessage}
-            <span style={{ fontWeight: 'normal' }}>
-              {' '}
-              (in {response.totalTime}ms)
-            </span>
-          </span>
-        );
-      } else if (searcher.state.error) {
-        // got an error...
-        message = `Error: ${searcher.state.error}`;
+    if (this.props.response) {
+      response = this.props.response;
+    } else {
+      const searcher = this.context.searcher;
+      if (searcher) {
+        response = searcher.state.response;
```

Comment:
This is a ninja modifier. You are modifying props here which is not immediately obvious, but breaks the Golden Rule.
On `line 45`, you are passing the reference of `this.props.response` to a new variable. Because of that, you are directly updating `this.props.response` on this line.

Reply:
I might be missing something, but line 45 is inside an `if` block, and line 49, where you are saying that I'm modifying the props, is in the else case, so line 49 would never be executed if `response` was set equal to props from line 45.

Reply:
Never mind - going with the `renderMessage` approach you suggest below makes this irrelevant anyway and your way is cleaner.